src/util/task_runner.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_TASK_RUNNER_H
+#define BITCOIN_UTIL_TASK_RUNNER_H
+
+#include <cstddef>
+#include <functional>
+
+namespace util {
+
+/** @file
+ * This header provides an interface and simple implementation for a task
+ * runner. Another threaded, serial implementation using a queue is available in
+ * the scheduler module's SerialTaskRunner.
+ */
+
+class TaskRunnerInterface
+{
+public:
+    virtual ~TaskRunnerInterface() {}
+
+    /**
+     * The callback can either be queued for later/asynchronous/threaded
+     * processing, or be executed immediately for synchronous processing. When
+     * used in the ValidationSignals synchronous processing will block
+     * validation. It is called for each subscriber on each validation
+     * interface event.
```

Comment:
https://github.com/bitcoin/bitcoin/pull/28960#discussion_r1492391457

Reply:
re: https://github.com/bitcoin/bitcoin/pull/28960#discussion_r1492409498

Yeah the sentence "When used in the ValidationSignals synchronous processing will block validation. It is called for each subscriber on each validation interface event." is a little misleading because only one task per event is inserted, regardless of number of subscribers. In the future we might want to change this so indexes/wallets don't block each other, but that is the current implementation.

I would fix this by deleting this sentence here, because this information is more relevant to ValidationSignals anyway than the TaskRunnerInterface class. A good place to move it would be to the ValidationSignals constructor:

https://github.com/bitcoin/bitcoin/blob/316c7c845036cbffa22b9e44f31dca8573ffb639/src/validationinterface.h#L166

And I like furszy's suggested wording https://github.com/bitcoin/bitcoin/pull/28960#discussion_r1492391457

Reply:
>  A good place to move it would be to the ValidationSignals constructor
I would rather place it above the imp class declaration, but just a tiny styling nit.